From 323e39a24e04de6305ce9c0a4ba38738389c333b Mon Sep 17 00:00:00 2001
From: Matthias Runge <mrunge@redhat.com>
Date: Fri, 4 Sep 2015 10:00:58 +0200
Subject: [PATCH] fix credentials boxes when selecting WebSSO

Change-Id: Id2cf2c9ef1e0232e5b4c8cb748c22420c42c0544
Resolves: rhbz#1247804
Reviewed-on: https://code.engineering.redhat.com/gerrit/57045
Tested-by: RHOS Jenkins <apevec+rhosci@redhat.com>
Reviewed-by: Matthias Runge <mrunge@redhat.com>
Tested-by: Matthias Runge <mrunge@redhat.com>
(cherry picked from commit 64d54492e46ea6d02a0cc107924cce674f47f98d)
---
 .../theme/templates/auth/_description.html         | 13 ++++++++++
 .../dashboards/theme/templates/auth/_login.html    | 29 +++++++++++++++++++---
 .../dashboards/theme/templates/auth/login.html     | 15 +++++++++++
 .../dashboards/theme/templates/splash.html         | 12 ++++++++-
 4 files changed, 64 insertions(+), 5 deletions(-)
 create mode 100644 openstack_dashboard/dashboards/theme/templates/auth/_description.html

diff --git a/openstack_dashboard/dashboards/theme/templates/auth/_description.html b/openstack_dashboard/dashboards/theme/templates/auth/_description.html
new file mode 100644
index 0000000..76a3c28
--- /dev/null
+++ b/openstack_dashboard/dashboards/theme/templates/auth/_description.html
@@ -0,0 +1,13 @@
+{% load i18n %}
+
+{% comment %}
+  This help text will only show up if websso is enabled
+  because websso introduces new authentication mechanisms.
+{% endcomment %}
+<p id="help_text">
+  {% block websso-help-text %}
+    {% blocktrans %}
+      If you are not sure which authentication method to use, contact your administrator.
+    {% endblocktrans %}
+  {% endblock %}
+</p>
\ No newline at end of file
diff --git a/openstack_dashboard/dashboards/theme/templates/auth/_login.html b/openstack_dashboard/dashboards/theme/templates/auth/_login.html
index a0845a8..638dd16 100644
--- a/openstack_dashboard/dashboards/theme/templates/auth/_login.html
+++ b/openstack_dashboard/dashboards/theme/templates/auth/_login.html
@@ -1,15 +1,27 @@
 {% extends "horizon/common/_modal_form.html" %}
 {% load i18n %}
-{% load url from future %}
 
 {% block modal-header %}{% trans "Log In" %}{% endblock %}
 {% block modal_class %}rhlogin{% if hide %}modal hide{% endif %}{% endblock %}
 
-{% block autocomplete %}{{ HORIZON_CONFIG.password_autocomplete }}{% endblock %}
 {% block form_action %}{% url 'login' %}{% endblock %}
+{% block ng_controller %}hzLoginCtrl{% endblock %}
+{% block autocomplete %}{{ HORIZON_CONFIG.password_autocomplete }}{% endblock %}
 
 {% block modal-body %}
-  <fieldset>
+
+  {% comment %}
+    These fake fields are required to prevent Chrome v34+ from autofilling form.
+  {% endcomment %}
+  {% if HORIZON_CONFIG.password_autocomplete != "on" %}
+  <div class="fake_credentials" style="display: none">
+    <input type="text" name="fake_email" value="" />
+    <input type="password" name="fake_password" value="" />
+  </div>
+  {%endif%}
+  {% include "auth/_description.html" %}
+
+  <fieldset hz-login-finder>
     {% if request.user.is_authenticated and 'next' in request.GET %}
     <div class="control-group clearfix error">
       <span class="help-inline"><p>{% trans "You don't have permissions to access:" %}</p>
@@ -18,6 +30,12 @@
         <a href="{% url 'horizon:user_home' %}">{% trans "home page" %}</a></p>
       </span>
     </div>
+    {% if request.COOKIES.logout_reason %}
+    <div class="form-group clearfix error" id="logout_reason">
+        <span class="help-block alert alert-danger"><p>{{ request.COOKIES.logout_reason }}</p></span>
+    </div>
+    {% endif %}
+
     {% endif %}
     {% if next %}<input type="hidden" name="{{ redirect_field_name }}" value="{{ next }}" />{% endif %}
     {% include "horizon/common/_form_fields.html" %}
@@ -25,5 +43,8 @@
 {% endblock %}
 
 {% block modal-footer %}
-  <button type="submit" class="btn btn-primary btn-large">{% trans "Sign In" %}</button>
+  <button id="loginBtn" type="submit" class="btn btn-primary btn-large">
+    <span ng-show="auth_type==='credentials'">{% trans "Sign In" %}</span>
+    <span ng-hide="auth_type==='credentials'" ng-cloak>{% trans "Connect" %}</span>
+  </button>
 {% endblock %}
diff --git a/openstack_dashboard/dashboards/theme/templates/auth/login.html b/openstack_dashboard/dashboards/theme/templates/auth/login.html
index ccfc71e..5ce49d8 100644
--- a/openstack_dashboard/dashboards/theme/templates/auth/login.html
+++ b/openstack_dashboard/dashboards/theme/templates/auth/login.html
@@ -1,2 +1,17 @@
 {% extends "splash.html" %}
 {% load i18n %}
+
+{% block title %}{% trans "Login" %}{% endblock %}
+{% include "horizon/_conf.html" %}
+{% include "horizon/client/side/_script_loader.html" %}
+{% include "horizon/_custom_head_js.html" %}
+{% block body_id %}splash{% endblock %}
+
+{% block content %}
+    {% include 'auth/_login.html' %}
+{% endblock %}
+
+{% block js %}
+    {% include "horizon/_scripts.html" %}
+{% endblock %}
+
diff --git a/openstack_dashboard/dashboards/theme/templates/splash.html b/openstack_dashboard/dashboards/theme/templates/splash.html
index 4b4e61a..87ce4b5 100644
--- a/openstack_dashboard/dashboards/theme/templates/splash.html
+++ b/openstack_dashboard/dashboards/theme/templates/splash.html
@@ -5,8 +5,13 @@
     <meta content='text/html; charset=utf-8' http-equiv='Content-Type' />
     <title>{% trans "Login" %} - {% site_branding %}</title>
     {% include "_stylesheets.html" %}
+
+    {% include "horizon/_conf.html" %}
+    {% include "horizon/client_side/_script_loader.html" %}
+    {% include "horizon/_custom_head_js.html" %}
+
   </head>
-  <body id="splash">
+  <body id="splash" ng-app='hz'>
     <a href="http://www.redhat.com/" id="redhat">
       <img src="/dashboard/static/dashboard/img/logo.svg" alt="Red Hat" />
     </a>
@@ -20,5 +25,10 @@
         {% include 'auth/_login.html' %}
       </div><!--/.row-fluid-->
     </div><!--/.container-->
+
+    {% block js %}
+      {% include "horizon/_scripts.html" %}
+    {% endblock %}
+
   </body>
 </html>
